src/app/docs/macro.js

```diff
@@ -15,25 +15,47 @@ angular
     $scope.project = projectService;
     $scope.codeService = codeService;
 
-    function getReferences(project, macro) {
-        var references = _.filter(project.nodes, function(node) {
+    function getReferences(project, self) {
```

Comment:
Do you think there's a way to reconcile this logic with the code in `src/app/docs/utils.js`? I totally get that the different namespaces for macros and nodes could make this challenging, but it feels funny to have two very similar sets of functions so far apart from each other. Let me know what you think!

Reply:
The decision to include these two functions in `macro.js` and separate from `utils.js` was to accommodate for the differing properties of macros and nodes. Would it make sense to you to have the two functions in `macro.js` included in `utils.js` as well?

Reply:
@drewbanin I moved the functions related to obtaining references and parents for macros from `macro.js` to `dag_utils.js` (previously utils.js). Let me know what you think.